### ISSUE TITLE
Ensure LIMIT pushdown tests use SUPPORTS_LIMIT_PUSHDOWN behaviour flag

### ISSUE THREAD

**Issue by losipiuk** [trinodb/trino#7430]

**Comment by losipiuk** [comment]
cc: @hashhar @findepi

**Comment by hashhar** [comment]
We'll first need common limit pushdown tests. The one in ATDQ only runs LIMIIT queries but doesn't verify pushdown.
Each connector copies that test and adds proper `isFullyPushedDown` calls.

IMO the common tests should go to `BaseConnectorTest` - since limit pushdown comes from ConnectorMetadata#applyLimit and should be tested for in all connectors?

Once we agree on where the tests should go I'll submit a PR.

**Comment by losipiuk** [comment]
Good points @hashhar . I did not really do my homework and verify if we have test coverage already in proper place. Thanks

**Comment by findepi** [comment]
We can rephrase this as "move/define/create common re-usable LIMIT pushdown tests"

> IMO the common tests should go to BaseConnectorTest - since limit pushdown comes from ConnectorMetadata#applyLimit and should be tested for in all connectors?

The tested combinations -- LIMIT pushdown with this, LIMIT pushdown with that, etc. -- are specific to situations where remote system is a relational database.
I would put the test in `BaseJdbcConnectorTest` as we did with TopN pushdown and Join pushdown.
But we can try be more generic and move these tests higher up, into `BaseConnectorTest`. (Then... would `BaseJdbcConnectorTest` become empty?)
(we can also try with LIMIT pushdown alone and see what happens)
